src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -7,23 +7,21 @@
 import com.dev.cinema.util.HibernateUtil;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
+import java.time.LocalTime;
 import java.util.List;
 import org.hibernate.Session;
 import org.hibernate.Transaction;
 import org.hibernate.query.Query;
 
 @Dao
 public class MovieSessionDaoImpl implements MovieSessionDao {
-    public static final int HOUR = 23;
-    public static final int MINUTE = 59;
 
     @Override
     public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
             LocalDateTime startOfDay = date.atStartOfDay();
-            LocalDateTime endOfDay = date.atTime(HOUR, MINUTE);
+            LocalDateTime endOfDay = date.atTime(LocalTime.MAX);
             Query<MovieSession> movieQuery = session.createQuery("from MovieSession ms "
-                    + "join fetch ms.movie "
```

Comment:
we can use join fetch to add `JOIN` to our query when we want to fully initialize the object when we have lazy fetching

you are eager-fetching by default for `@ManyToOne` relationship

Reply:
yes, agree with you